Dockerfile

```diff
@@ -1,14 +1,24 @@
 FROM python:3.10-slim
 
+ARG BASHIO_VERSION="v0.16.2"
+
 WORKDIR /app
 
 COPY requirements.txt ./
 
 RUN apt-get update && \
-  apt-get install -y --no-install-recommends gcc python3-dev libssl-dev libxml2-dev libxslt-dev python3-dev jq && \
+  apt-get install -y --no-install-recommends curl tar gcc python3-dev libssl-dev libxml2-dev libxslt-dev python3-dev jq && \
   pip3 install -r requirements.txt && \
   apt-get remove -y gcc python3-dev libssl-dev && \
-  apt-get autoremove -y
+  apt-get autoremove -y \
+    && curl -J -L -o /tmp/bashio.tar.gz \
```

Comment:
Think that's currently not possible because i can´t found a sha file or something else for the release archive. The file must provided from the repo owner. In the past, there was a content-md5 header but the header was removed.

https://github.com/orgs/community/discussions/23512#discussioncomment-7856465

Reply:
I just copied the above lines from [dockerfile ](https://github.com/hassio-addons/addon-debian-base/blob/3e035aaf8a0d878bd7e62b9c1b264f21ef1c6a91/base/Dockerfile#L65C1-L73C6) of "Home Assistant Community Add-on: Debian Base Images". If we want to do a checksum check I'll have to look into it...

Reply:
> Think that's currently not possible because i can´t found a sha file or something else for the release archive. The file must provided from the repo owner. In the past, there was a content-md5 header but the header was removed.
> 
> https://github.com/orgs/community/discussions/23512#discussioncomment-7856465

It doesn't necessarily need to be a published value, we can just sha256sum a copy we download. Mostly just to protect against release files being replaced in future.

Reply:
Since we download a specific version of bashio, I have added a SHA256 check as suggested by @Meatballs1